Fix DEPLOY_ENV's default behaviour (#75)
mapbox/underreact: README.md

### `DEPLOY_ENV` and `NODE_ENV`

- `NODE_ENV` will default to `'development'` in [development mode](#modes) and `'production'` in [production mode](#modes).
- It's recommended that you *not* set `NODE_ENV` manually: use Underreact modes instead. But if you do set `NODE_ENV`, it must be `'development'`, `'production'`, or `'test'`.

> - By default, `DEPLOY_ENV` is set to `'development'` in both development and production mode, but you can set it to any value you wish to better align with your target environments (e.g. `DEPLOY_ENV=something npx underreact build`). For example, you may want to set it to `'staging'` when building for a staging environment, or `'test'` when testing.

Reviewer: ```suggestion
- By default, `DEPLOY_ENV` is set to `'development'`. You can it to any value you wish to better align with your target environments (e.g. `DEPLOY_ENV=something npx underreact build`) and this value will be made available in your client-side code on `process.env.DEPLOY_ENV`. For example, you may want to set it to `'staging'` when building for a staging environment, `'production'` when building for production, or `'test'` when testing.
```
